special-pages/pages/new-tab/app/components/CompanyIcon.js

```diff
@@ -0,0 +1,78 @@
+import styles from './CompanyIcon.module.css';
+import { DDG_STATS_OTHER_COMPANY_IDENTIFIER } from '../privacy-stats/constants.js';
+import { h } from 'preact';
+import { useState } from 'preact/hooks';
+
+const mappings = {
+    'google-analytics-google': 'google-analytics',
+};
+
+const states = /** @type {const} */ ({
+    loading: 'loading',
+    loaded: 'loaded',
+    loadingFallback: 'loadingFallback',
+    loadedFallback: 'loadedFallback',
+    errored: 'errored',
+});
+
+/**
+ * @typedef {states[keyof states]} State
+ */
+
+/**
+ * @param {object} props
+ * @param {string} props.displayName
+ */
+export function CompanyIcon({ displayName }) {
+    const icon = displayName.toLowerCase().split('.')[0];
+    const cleaned = icon.replace(/[^a-z ]/g, '').replace(/ /g, '-');
+    const id = cleaned in mappings ? mappings[cleaned] : cleaned;
+    const firstChar = id[0];
+    const [state, setState] = useState(/** @type {State} */ (states.loading));
+
+    const src =
+        state === 'loading' || state === 'loaded'
+            ? `./company-icons/${id}.svg`
+            : state === 'loadingFallback' || state === 'loadedFallback'
+              ? `./company-icons/${firstChar}.svg`
+              : null;
+
+    if (src === null || icon === DDG_STATS_OTHER_COMPANY_IDENTIFIER) {
+        return (
+            <span className={styles.icon}>
+                <Other />
+            </span>
+        );
+    }
+
+    return (
+        <span className={styles.icon}>
+            <img
+                src={src}
+                alt={''}
+                class={styles.companyImgIcon}
+                data-loaded={state === states.loaded || state === states.loadedFallback}
+                onLoad={() => setState((prev) => (prev === states.loading ? states.loaded : states.loadedFallback))}
+                onError={() => {
+                    setState((prev) => {
+                        if (prev === states.loading) return states.loadingFallback;
+                        return states.errored;
+                    });
+                }}
+            />
+        </span>
+    );
+}
```

Comment:
This was just moved to a separate file so it can be used for Activity feed in another PR :)